book/plugins/general-abilities/video-content-download/ios/video-content-download-ios.md

```diff
@@ -0,0 +1,156 @@
+				guard let model = self.componentDataSourceModel as? ZPDownloadableItemProtocol else {
+						return
+				}
+
+				switch state {
+						case .error:
+								//as an example, cancel downloading if error received
+								ZAAppConnector.sharedInstance().hqmeDelegate?.cancelDownloading(model)
+						default:
+								break
+				}
+		}
+		```
+
+	- `func downloadButton(_ downloadButton: ZPDownloadButtonProtocol, tappedWithState state: ZPDownloadButtonState)`
+		- fired on download button click, to implement the action on each one of the download button states.
+
+		- implementation example:
+
+		```swift
+		public func downloadButton(_ downloadButton: ZPDownloadButtonProtocol, tappedWithState state: ZPDownloadButtonState) {
+		    guard let model = self.componentDataSourceModel as? ZPDownloadableItemProtocol else {
+		        return
+		    }
+
+		    switch state {
+		    case .downloaded:
+						//perform an action on already downloaded item. (ex. remove)
+		        break
+		    case .startDownload:
+						//perform an action on not downloaded item
+						ZAAppConnector.sharedInstance().hqmeDelegate?.download(model)
+
+						//update state to pending
+						downloadButton.downloadStateChange(to: .pending)
+		        break
+		    case .downloading:
+						//perform an action on item that is currently being downloaded
+		        ZAAppConnector.sharedInstance().hqmeDelegate?.cancelDownloading(model)
+
+		        //update state to initial after cleanup
+		        downloadButton.downloadStateChange(to: .startDownload)
+		        break
+		    case .error:
+						//perform an action on item that got an error in downloading. (ex. remove)
+		        break
+		    default:
+		        break
+		    }
+	  }
+		```
+
+	- `func downloadStateChangeNotificationName() -> String?`
+		- Defines the unique StateChange notification name for download item based on item identifier
+			- example: "AssetDownloadStateChanged"+identifier.md5()
+
+	- `func downloadingProgressChangeNotificationName() -> String?`
+		- Defines the unique ProgressChange notification name for download item based on item identifier
+			- example: "AssetDownloadProgressChanged"+identifier.md5()
+
+	- `func downloadButtonCustomImagesSuffix() -> String?`
+		- Defines the custom images suffix for download button download states images.
+		There are base sdk images for download button different states (explained in images section), custom suffix will be added to the image name lookup.
+		example:
+			 - item not downloaded image is `generic_item_download_btn` (value with custom suffix XYZ `generic_item_download_btnXYZ`)
+			 - item download progress image `generic_item_downloading_btn_1` (value with custom suffix XYZ `generic_item_downloading_btnXYZ_1`)
+
+2. Create a download button container on screen.
+
+3. Add download button to the container
+
+	- implementation example:
+
+	```swift
```

Comment:
can you clean up this sample from "fluff" to minimal real world necessities ?
There's really no need for the constraints and stuff like that.